src/Utils/obiDialogParser.ts

```diff
@@ -181,39 +207,89 @@ export class ObiDialogParser {
         }
         return trainDialogs
     }
-    
-    private async getScorerStepfromActivity(prompt: string): Promise<CLM.TrainScorerStep> {
-    
+
+    private async getScorerStepFromActivity(prompt: string): Promise<CLM.TrainScorerStep> {
         const parsedActivity = prompt.substring(prompt.indexOf("[") + 1, prompt.lastIndexOf("]")).trim()
         let response = this.composerDialog.lgMap.get(parsedActivity)
         if (!response) {
             // LARS thow error once CCI .dialog transformer has been fixed
-            response = { text: "Can't Parse LG", suggestions: []}
-           //throw new Error(`LG name ${prompt} undefined`)
+            response = { text: "Can't Parse LG", suggestions: [] }
+            //throw new Error(`LG name ${prompt} undefined`)
         }
-    
-        let scoredAction: CLM.ScoredAction | undefined
 
         let scoreInput: CLM.ScoreInput = {
-            filledEntities: [],  //LARS handle filled entities from api calls
+            filledEntities: [],
             context: {},
             maskedActions: []
         }
-    
+
         const importText = response.suggestions.length > 0 ? JSON.stringify(response) : response.text
         return {
             importText,
             input: scoreInput,
             labelAction: CLM.CL_STUB_IMPORT_ACTION_ID,
             logicResult: undefined,  // LARS handle api calls
+            scoredAction: undefined
+        }
+    }
+
+    /**
+     * HttpRequest represents a RESTful request with known input and output parameters.
+     * The .dialog file is expected to have a field `responseFields` that enumerates the top-level
+     * output parameters of the response object.  Note that as of 2019.09, this field is specific
+     * to ConversationLearner and is not part of the OBI spec.
+     */
+    private async createActionFromHttpRequest(step: OBIDialog, nextStep: OBIDialog | undefined):
+        Promise<CLM.TrainScorerStep> {
+        if (!step.url) {
+            throw new Error('HTTP requests require url')
+        }
+        const isTerminal = (!nextStep || nextStep.$type === OBIStepType.TEXT_INPUT ||
```

Comment:
@larsliden does this logic look correct ?

Reply:
Thanks for making it `HttpRequest` instead of `HTTPRequest`

Reply:
Probably more complicated.  This is looking only at the current node to check for a next step, but the next step could actually be coming from a node further down in the tree.  Have to think about the best way to do this.  Might be easiest to keep a cache of actions to create and then walk the train dialogs afterwards to determine terminal state.  I'd suggest not blocking check-in of what you have so far, but let's chat and figure out the best approach